File: frontend/src/app/models/order.ts
export interface CreateOrder {
  addressId: number;
  paymentMethod: string;
}

export interface OrderItem {
  productId: number;
  productName: string;
  imageUrl?: string;
  quantity: number;
  price: number;
  subTotal: number;
}
export interface OrderAddress {
  fullName: string;
  phone: string;
  addressLine1: string;
  addressLine2?: string;
  landmark?: string;
  city: string;
  state: string;
  postalCode: string;
  country: string;
}

export interface Order {
  orderId: number;
  orderDate: string;
  totalAmount: number;
  orderStatus: string;
  paymentStatus: string;
  paymentMethod: string;
  addressId: number;
  shippingAddress?: OrderAddress;
  items: OrderItem[];
}

File: freshcart/Models/Order.cs
namespace freshcart.Models
{
    public class Order
    {
        public int OrderId { get; set; }

        public int UserId { get; set; }

        public int AddressId { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.Now;

        public decimal TotalAmount { get; set; }

        public string OrderStatus { get; set; } = "Pending";

        public string PaymentStatus { get; set; } = "Pending";

        public string PaymentMethod { get; set; } = "CashOnDelivery";

        public User? User { get; set; }

        public Address? Address { get; set; }

        public ICollection<OrderItem> OrderItems { get; set; }
            = new List<OrderItem>();
    }
    public class OrderItem
    {
        public int OrderItemId { get; set; }

        public int OrderId { get; set; }

        public int ProductId { get; set; }

        public int Quantity { get; set; }

        public decimal Price { get; set; }

        public decimal SubTotal { get; set; }

        public Order? Order { get; set; }

        public Product? Product { get; set; }
    }
}
